crates/matrix-sdk-indexeddb/src/crypto_store/migrations/mod.rs

```diff
@@ -28,6 +30,37 @@ mod v7;
 mod v7_to_v8;
 mod v8_to_v10;
 
+struct MigrationDb {
+    db: IdbDatabase,
+    next_version: u32,
+}
```

Comment:
Good idea! It's even better with this approach because if an error occurs, we are sure the DB is closed.